hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ClientManagerImpl.java

```diff
@@ -128,6 +143,17 @@ private CallableRaisingIOE<S3AsyncClient> createAyncClient() {
         () -> clientFactory.createS3AsyncClient(getUri(), clientCreationParameters));
   }
 
+  /**
+   * Create the function to create the unencrypted S3 client.
+   * @return a callable which will create the client.
+   */
+  private CallableRaisingIOE<S3Client> createUnencryptedS3Client() {
+    return trackDurationOfOperation(
+        durationTrackerFactory,
+        STORE_CLIENT_CREATION.getSymbol(),
+        () -> unencryptedClientFactory.createS3Client(getUri(), clientCreationParameters));
```

Comment:
can this throw a NPE? It looks like unencryptedClientFactory is null when not using `CSEV1CompatibleS3AFileSystemHandler`, what happens then?